### PR TITLE
uPnL and unwind price

### PR DESCRIPTION
**Pull request by 0xZenus**

### CODE REVIEW

File: products/dated-irs/src/libraries/ExposureHelpers.sol [hunk 1]
```diff
@@ -59,55 +60,58 @@ library ExposureHelpers {
...
+        int256 baseBalance,
+        int256 quoteBalance,
+        int256 uPnL
+    ) internal view returns (SD59x18 unwindPrice) {
+        SD59x18 timeDeltaAnnualized = Time.timeDeltaAnnualized(maturityTimestamp).intoSD59x18();
+
+        Market.Data storage market = Market.exists(marketId);
+        UD60x18 currentLiquidityIndex = market.getRateIndexCurrent();
+
+        int256 notional = mulUDxInt(currentLiquidityIndex, baseBalance);
```
Comment (avniculae) [inline products/dated-irs/src/libraries/ExposureHelpers.sol:105]: This does not work with linear rate oracles. Please use baseToNotional or baseToAnnualisedNotional functions

File: products/dated-irs/src/libraries/ExposureHelpers.sol [hunk 2]
```diff
@@ -59,55 +60,58 @@ library ExposureHelpers {
...
+
         UD60x18 twap = IPool(poolAddress).getAdjustedDatedIRSTwap(
             marketId, 
             maturityTimestamp, 
             orderSizeWad, 
             market.marketConfig.twapLookbackWindow
         );
 
-        unwindQuote = mulUDxInt(twap.mul(timeDeltaAnnualized).add(UNIT), mulUDxInt(currentLiquidityIndex, baseAmount));
+        int256 unwindQuote = mulUDxInt(twap.mul(timeDeltaAnnualized).add(UNIT_ud), mulUDxInt(currentLiquidityIndex, baseBalance));
```
Comment (avniculae) [inline products/dated-irs/src/libraries/ExposureHelpers.sol:88]: Is `mulUDxInt(currentLiquidityIndex, baseBalance)` the notional? if yes, this does not work with linear rate oracle, see my other comment

File: products/dated-irs/src/libraries/ExposureHelpers.sol (hunk 1 above)
Comment (0xZenus) [reply products/dated-irs/src/libraries/ExposureHelpers.sol:105]: great catch!

File: products/dated-irs/src/libraries/ExposureHelpers.sol (hunk 2 above)
Comment (0xZenus) [reply products/dated-irs/src/libraries/ExposureHelpers.sol:88]: yes, great catches! thanks!

File: products/dated-irs/src/libraries/ExposureHelpers.sol (hunk 1 above)
Comment (0xZenus) [reply products/dated-irs/src/libraries/ExposureHelpers.sol]: ab0b24d2ce47e323e62534999e221aa40faa13b8

File: products/dated-irs/src/libraries/ExposureHelpers.sol (hunk 2 above)
Comment (0xZenus) [reply products/dated-irs/src/libraries/ExposureHelpers.sol]: ab0b24d2ce47e323e62534999e221aa40faa13b8

**Review by avniculae — APPROVED** [review]

**Pull request merged**